### DISCUSSION TITLE
Multi-tenant site, sharded database, distinguished by hostname (subdomain), users in identity table per db, supported?

### DISCUSSION THREAD

**Discussion by Phlow2001** [JonPSmith/AuthPermissions.AspNetCore#75] [Q&A]
I'm trying to determine if this library can support our use case.

We are aiming for having a multi-tenant website(1), where we send wildcard subdomain traffic in, and use the hostname to determine the tenant.
Each tenant has its own SQL DB, with the same schema(2), including its own asp.net core identity and ef migration tables.  So no central table that includes tenant users.

Is this something that's possible?  Or could be implemented without much work, and still take advantage of the features of this library?

(1) there's also another sub-application that hosts a different asp.net core web app, at a virtual path, that talks to the same tenant DB, but different identity tables.
(2) we will want to support gradual migration of tenant db version, so the site will have to stop serving certain tenants while they're being migrated, and a new site version will need to pick them up

**Comment by JonPSmith** [comment]
Hi @Phlow2001,

I don't think the AuthP library is going to help you because AuthP uses the logged-in user's ID to lookup the tenant the user is using. 

You might find some of the articles I have written about multi-tenant apps useful. You can see the full list of the articles at the top left of the [repo's documentation home page](https://github.com/JonPSmith/AuthPermissions.AspNetCore/wiki).
